### PR TITLE
Upgrade to mastarm 5.3.2

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
This PR upgrades mastarm to 5.3.2. As a result, new code style rules (prop sorting, object key sorting, and destructuring variable declarations) are enforced during pre-commit AND during CI build.

The changes to the files in this PR only make sure the CI build passes. No other changes are made.

Some checks are disabled inline for the time being:
- Where the order of the key matters (e.g. in enumerations)
- a11y checks regarding `<label>` `for` attribute and regarding `<div>` with `onClick` events.

### CODE REVIEW

**Comment by binh-dam-ibigroup** [comment]
Okay now the new eslint rules are in full force in mastarm 5.3.2, so I'm reassigning to myself.

**Review by landonreed — APPROVED** [review]
If no other changes were made other than automatic lint fixes, I suppose we can move forward with this.

File: lib/components/app/call-taker-panel.js [hunk 1]
```diff
@@ -158,15 +158,15 @@ class CallTakerPanel extends Component {
...
+                  inputPlaceholder={`Enter intermediate destination`}
+                  key={i}
                   location={place}
-                  onLocationCleared={this._removePlace}
                   // FIXME: function def
-                  onLocationSelected={result => this._addPlace(result, i)}
-                  // FIXME: allow intermediate location type.
                   locationType='to'
-                  inputPlaceholder={`Enter intermediate destination`}
+                  // FIXME: allow intermediate location type.
```
Comment (evansiroky) [inline lib/components/app/call-taker-panel.js:167]: Comment is now out-of-place.

File: lib/components/form/call-taker/advanced-options.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable jsx-a11y/label-has-for */
```
Comment (evansiroky) [inline lib/components/form/call-taker/advanced-options.js:1]: Can you add a FIXME for this file-wide disable?

File: lib/components/form/date-time-modal.js [hunk 3]
```diff
@@ -28,15 +28,15 @@ class DateTimeModal extends Component {
...
             date={date}
+            dateFormatLegacy={dateFormatLegacy}
             departArrive={departArrive}
             onQueryParamChange={setQueryParam}
-            time={time}
             // These props below are for Safari on MacOS, and legacy browsers
             // that don't support `<input type="time|date">`.
             // These props are not relevant in modern browsers,
             // where `<input type="time|date">` already
             // formats the time|date according to the OS settings.
```
Comment (evansiroky) [inline lib/components/form/date-time-modal.js:38]: Comment is now out-of-date since it appeared to also apply to `dateFormatLegacy`.

File: lib/components/form/date-time-modal.js [hunk 4]
```diff
@@ -46,16 +46,17 @@ class DateTimeModal extends Component {
...
   return {
     config,
-    departArrive,
     date,
+    departArrive,
     time,
     // These props below are for legacy browsers (see render method above).
-    timeFormatLegacy: coreUtils.time.getTimeFormat(config),
-    dateFormatLegacy: coreUtils.time.getDateFormat(config)
+    // eslint-disable-next-line sort-keys
```
Comment (evansiroky) [inline lib/components/form/date-time-modal.js:57]: Yeah, this should be used in https://github.com/opentripplanner/otp-react-redux/pull/390/files#r666591987.

File: lib/components/form/date-time-preview.js [hunk 5]
```diff
@@ -1,3 +1,5 @@
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
```
Comment (evansiroky) [inline lib/components/form/date-time-preview.js:2]: Add FIXME for file-wide disables.

File: lib/components/form/settings-preview.js [hunk 6]
```diff
@@ -1,22 +1,26 @@
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
```
Comment (evansiroky) [inline lib/components/form/settings-preview.js:2]: Add FIXME for file-wide eslint disable

File: lib/components/form/styled copy.js [hunk 7]
```diff
@@ -0,0 +1,184 @@
+import styled, { css } from 'styled-components'
```
Comment (evansiroky) [inline lib/components/form/styled copy.js:1]: Duplicate file??

File: lib/components/form/trimet.styled.js [hunk 8]
```diff
@@ -0,0 +1,147 @@
+import React from 'react'
```
Comment (evansiroky) [inline lib/components/form/trimet.styled.js:1]: Why is this file being created?

File: lib/components/map/leg-diagram.js [hunk 9]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable jsx-a11y/no-static-element-interactions */
```
Comment (evansiroky) [inline lib/components/map/leg-diagram.js:1]: Add FIXME to remove this file-wide disable rule.

File: lib/components/form/styled copy.js (hunk 7 above)
Comment (binh-dam-ibigroup) [reply lib/components/form/styled copy.js:1]: Probably it got in doing a bulk git add, it shouldn't be in here.

File: lib/components/form/trimet.styled.js (hunk 8 above)
Comment (binh-dam-ibigroup) [reply lib/components/form/trimet.styled.js:1]: Probably it got in doing a bulk git add, it shouldn't be in here.

File: lib/components/app/call-taker-panel.js (hunk 1 above)
Comment (binh-dam-ibigroup) [reply lib/components/app/call-taker-panel.js:165]: Good catch, thanks, fixed in https://github.com/opentripplanner/otp-react-redux/pull/390/commits/1a526dd2886f67b837dfe118666f9d9d719988f5.

File: lib/components/form/date-time-modal.js (hunk 3 above)
Comment (binh-dam-ibigroup) [reply lib/components/form/date-time-modal.js]: Good catch, fixed in https://github.com/opentripplanner/otp-react-redux/pull/390/commits/1a526dd2886f67b837dfe118666f9d9d719988f5.

File: lib/components/form/call-taker/advanced-options.js (hunk 2 above)
Comment (binh-dam-ibigroup) [reply lib/components/form/call-taker/advanced-options.js:2]: I added the requested FIXMEs for rule exceptions in https://github.com/opentripplanner/otp-react-redux/pull/390/commits/1a526dd2886f67b837dfe118666f9d9d719988f5.

File: lib/components/narrative/default/default-itinerary.js [hunk 10]
```diff
@@ -210,14 +210,14 @@ class DefaultItinerary extends NarrativeItinerary {
     return (
       <div
         className={`option default-itin${active ? ' active' : ''}${expanded ? ' expanded' : ''}`}
-        role='presentation'
+        onMouseEnter={this._onMouseEnter}
         // FIXME: Move style to css
```
Comment (evansiroky) [inline lib/components/narrative/default/default-itinerary.js:214]: Comment out of place

File: lib/components/narrative/realtime-annotation.js [hunk 11]
```diff
@@ -63,15 +63,15 @@ export default class RealtimeAnnotation extends Component {
...
         overlay={
-          <Popover style={{maxWidth: '300px'}} id='popover-positioned-bottom'>
+          <Popover id='popover-positioned-bottom' style={{maxWidth: '300px'}}>
             {innerContent}
           </Popover>
-        }>
+        }
+        placement='bottom'
+        // container={this}
+        // containerPadding={40}
```
Comment (evansiroky) [inline lib/components/narrative/realtime-annotation.js:73]: Are these comments still needed?

File: lib/components/viewers/trip-viewer.js [hunk 12]
```diff
@@ -1,16 +1,17 @@
+/* eslint-disable jsx-a11y/label-has-for */
```
Comment (evansiroky) [inline lib/components/viewers/trip-viewer.js:1]: Add FIXME

File: lib/util/api.js [hunk 13]
```diff
@@ -0,0 +1,15 @@
+if (typeof (fetch) === 'undefined') require('isomorphic-fetch')
```
Comment (evansiroky) [inline lib/util/api.js:1]: Is this file needed?

**Review by evansiroky — CHANGES_REQUESTED** [review]
Finally got through every file, have a few more comments.

File: lib/util/api.js (hunk 13 above)
Comment (binh-dam-ibigroup) [reply lib/util/api.js]: It is not.

File: lib/components/viewers/trip-viewer.js (hunk 12 above)
Comment (binh-dam-ibigroup) [reply lib/components/viewers/trip-viewer.js:2]: Added in https://github.com/opentripplanner/otp-react-redux/pull/390/commits/11773ca2755c074d3ca83db0e40a92031c737b97.

File: lib/components/narrative/realtime-annotation.js (hunk 11 above)
Comment (binh-dam-ibigroup) [reply lib/components/narrative/realtime-annotation.js]: They seem from a different era, I removed them in https://github.com/opentripplanner/otp-react-redux/pull/390/commits/11773ca2755c074d3ca83db0e40a92031c737b97.

File: lib/components/narrative/default/default-itinerary.js (hunk 10 above)
Comment (binh-dam-ibigroup) [reply lib/components/narrative/default/default-itinerary.js]: Moved in https://github.com/opentripplanner/otp-react-redux/pull/390/commits/11773ca2755c074d3ca83db0e40a92031c737b97.

**Comment by binh-dam-ibigroup** [comment]
Thanks @evansiroky for the scrutinity, the number of files makes it easy to loose track of changes.

**Review by evansiroky — APPROVED** [review]
